controller/replicator.go

```diff
@@ -312,6 +312,95 @@ func (r *replicator) Snapshot(name string, userCreated bool, created string) err
 	return nil
 }
 
+func (r *replicator) GetLastSnapshot() (string, error) {
+	retErrorLock := sync.Mutex{}
+	retError := &BackendError{
+		Errors: map[string]error{},
+	}
+	replicaChains := make(map[string][]string)
+	wg := sync.WaitGroup{}
+	var success int
+
+	for addr, backend := range r.backends {
+		if backend.mode == types.RW {
+			wg.Add(1)
+			go func(address string, backend types.Backend) {
+				if chain, err := backend.GetReplicaChain(); err != nil {
+					logrus.Infof("failed taking snapshot at %s with err %v", addr, err)
+					retErrorLock.Lock()
+					retError.Errors[address] = err
+					retErrorLock.Unlock()
+				} else {
+					retErrorLock.Lock()
+					replicaChains[addr] = chain
+					success = success + 1
+					retErrorLock.Unlock()
+				}
+				wg.Done()
+			}(addr, backend.backend)
+		} else {
+			logrus.Infof("not getting chain from %s as replica in %v mode", addr, backend.mode)
+		}
+	}
+	wg.Wait()
+	if len(retError.Errors) != 0 {
+		return "", retError
+	}
+	if len(replicaChains) != len(r.backends) {
+		return "", fmt.Errorf("Not able to fetch chain from all replicas")
+	}
+	var snapName string
+	for _, chain := range replicaChains {
+		if len(chain) <= 1 {
```

Comment:
at one replica being <=1 and at other replicas being > 1 is possible. Instead of 'continue' doing 'break' is better?

Reply:
and by setting snapName to ""

Reply:
or return error also if caller can't handle empty snapName and nil error

Reply:
updating to return error. This was done primarily to handle single replica case. But yes this was a bug.